examples/js/exporters/GLTFExporter.js

```diff
@@ -277,53 +330,11 @@ THREE.GLTFExporter.prototype = {
 
 		}
 
-		/**
-		 * Get the required size + padding for a buffer, rounded to the next 4-byte boundary.
-		 * https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#data-alignment
-		 *
-		 * @param {Integer} bufferSize The size the original buffer.
-		 * @returns {Integer} new buffer size with required padding.
-		 *
-		 */
-		function getPaddedBufferSize( bufferSize ) {
-
-			return Math.ceil( bufferSize / 4 ) * 4;
-
-		}
-
-		/**
-		 * Returns a buffer aligned to 4-byte boundary.
-		 *
-		 * @param {ArrayBuffer} arrayBuffer Buffer to pad
-		 * @param {Integer} paddingByte (Optional)
-		 * @returns {ArrayBuffer} The same buffer if it's already aligned to 4-byte boundary or a new buffer
-		 */
-		function getPaddedArrayBuffer( arrayBuffer, paddingByte ) {
-
-			paddingByte = paddingByte || 0;
-
-			var paddedLength = getPaddedBufferSize( arrayBuffer.byteLength );
-
-			if ( paddedLength !== arrayBuffer.byteLength ) {
-
-				var array = new Uint8Array( paddedLength );
-				array.set( new Uint8Array( arrayBuffer ) );
-
-				if ( paddingByte !== 0 ) {
-
-					for ( var i = arrayBuffer.byteLength; i < paddedLength; i ++ ) {
-
-						array[ i ] = paddingByte;
+		function getFileNameFromUri( uri ) {
 
-					}
-
-				}
-
-				return array.buffer;
-
-			}
-
-			return arrayBuffer;
+			var parts = uri.split( '#' ).shift().split( '?' ).shift().split( '/' ).pop().split( '.' );
+			parts.pop();
+			return parts.join();
```

Comment:
For an absolute URL, I think just returning the filename like this is the best we can do. But what if the image source were a relative path like `hero/diffuse.png`? Or if another texture were referenced as `enemy/diffuse.png`? Here, that will work in "embedded" mode, but might be a problem when you try to write the "gltf" mode result to disk.

Reply:
Is there something better we can do when exporting in "gltf" mode? How can we determine the base path for these relative image paths?

Reply:
We're removing the mode where the glTF asset references absolute URLs, right? I think we shouldn't worry about preserving any relative path structure – just flatten all the images into `textures/` or the root folder, and deal with name collisions by appending an index.

Reply:
I believe my most recent change accomplishes this well enough. The collisions could be handled more gracefully, but it seems ok to always append an index.